### DISCUSSION TITLE
Head in the child component

### DISCUSSION THREAD

**Discussion by bbonch** [vercel/next.js#38564] [Help]
Here [docs](https://nextjs.org/learn/basics/assets-metadata-css/polishing-layout) you use Head in Layout component. And you use Head in index.js. Why? How is it possible to have such an error in the official documentation.

**Comment by icyJoseph** [comment]
Hi,

The `Head` component from `next/head`, not to confused with the `Head` from `next/document` is intended for this kind of usage.

Let's go through the link you shared:

```jsx
      <Head>
        <link rel="icon" href="/favicon.ico" />
        <meta
          name="description"
          content="Learn how to build a personal website using Next.js"
        />
        <meta
          property="og:image"
          content={`https://og-image.vercel.app/${encodeURI(
            siteTitle,
          )}.png?theme=light&md=0&fontSize=75px&images=https%3A%2F%2Fassets.vercel.com%2Fimage%2Fupload%2Ffront%2Fassets%2Fdesign%2Fnextjs-black-logo.svg`}
        />
        <meta name="og:title" content={siteTitle} />
        <meta name="twitter:card" content="summary_large_image" />
      </Head>
```

You see here, this is part of a Layout component, and it is setting a few things:

- favicon
- description meta tag
- open graph image
- open graph title, site title
- twitter card settings

However, it doesn't set the `<title>` tag, which is what the tab is titled after, among other SEO things. You can manually set this with `document.title` but then non JS entities won't see it. 

Now we see the `index` page:

```jsx
  <Head>
        <title>{siteTitle}</title>
  </Head>
```

This one takes care of setting the title. Often titles depend on the route you are at, and might even depend on user actions, etc. 

Since these are all injecting different things to the `head` of the HTML document, we are all good.

### default seo and nesting

The above is actually a cool feature, say we have this `_app`:

```jsx
import '@/styles/global.css';
import type { AppProps } from 'next/app';
import Head from 'next/head';

import { AppFooter } from '@/layouts/AppFooter';
import { AppHeader } from '@/layouts/AppHeader';

const MyApp = ({ Component, pageProps, router }: AppProps) => {
  return (
    <>
      <Head>
        <title>Default Title</title>
      </Head>

      <AppHeader />

      <Component {...pageProps} key={router.route} />

      <AppFooter />
    </>
  );
};

export default MyApp;
```

This means, that, for all pages, we will have a title called `Default Title`, and yet, we can, at page level, override that by:

```jsx
const Home: NextPage = () => {
  return (
    <div>
      <Head>
        <title>Home Page</title>
      </Head>

      Home
    </div>
  );
};

export default Home;
```

I don't want to overburden you with this things, but at the end of the day these are React components, and that's _**just JavaScript**_ as the saying goes. 

The docs are not doing anything particularly wrong, they are just showing you that you can place these in many places, and create patterns with them.

**Reply by bbonch** [reply]
They both import from next/head. Head is not head, it's just a component, so it can be anywhere. But not with this one. It should be at there root, _app or page component to override as you suggested. But Head can't be in an ordinary component. You should remove Head from Layout component.
